components/logins/src/db.rs

```diff
@@ -856,6 +856,44 @@ impl LoginDb {
         )?)
     }
 
+    pub fn potential_dupes_ignoring_username(&self, login: &Login) -> Result<Vec<Login>> {
+        // Could be lazy_static-ed...
+        lazy_static::lazy_static! {
+            static ref DUPES_IGNORING_USERNAME_SQL: String = format!(
+                "SELECT {common_cols} FROM loginsL
+                WHERE is_deleted = 0
+                    AND hostname = :hostname
+                    AND (
+                        formSubmitURL = :form_submit
```

Comment:
In https://github.com/mozilla/application-services/blame/master/components/logins/src/db.rs#L404 you're doing something subtly different, why is that?

Reply:
1. The way we dedupe during syncs is necessarially different. `find_dupe` is only used during sync. I think there's a patch somewhere that renames this to make it clearer but there are... many dead patches around duping. in logins (see 2)
2. The story around logins dupes is... Storied. TLDR that code is wrong, but it's tricky to fix. We have semi-regular meetings talking about this and other duping-related issues.

You'll note that this is in-line with how we detect dupes https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L819.